lib/modules/dosomething/dosomething_signup/dosomething_signup.module

```diff
@@ -243,6 +243,13 @@ function dosomething_signup_create($nid, $uid = NULL, $source = NULL, $timestamp
     return FALSE;
   }
 
+  if (module_exists('stathat')) {
+    // The source is often a string longer than just "reportback", this logic accounts for that.
```

Comment:
> The source is often a string longer than just "reportback"
How longer? Should it better clarify that the source string starts with `reportback`?